### PR TITLE
Add shadow to the base stylesheet

### PR DESCRIPTION
**Pull request by cristianames92**
This PR intents to add the shadow configurations to the base stylesheet

### CODE REVIEW

**Comment by guidomb** [comment]
@cristianames92 please add screenshots examples based on the Portal example project.

File: Portal/View/Component.swift [hunk 1]
```diff
@@ -141,8 +141,8 @@ extension Component {
         case .mapView(let properties, let style, let layout):
             return .mapView(properties, style, layout)
 
-        case .imageView(let image, let style, let layout):
-            return .imageView(image, style, layout)
+        case .imageView(let image, let clipToBounds, let style, let layout):
```
Comment (guidomb) [inline Portal/View/Component.swift]: Why are you adding this in this PR? It does not seem related to adding shadows? Also components should not have more than 3 associated values. Please sent a new PR adding this property as part of a `ImageProperties` struct which contains both `image` and `cliptToBounds` properties.

File: Portal/View/Shadow.swift [hunk 2]
```diff
@@ -0,0 +1,53 @@
+//
+//  Shadow.swift
+//  Portal
+//
+//  Created by Cristian Ames on 7/11/17.
+//  Copyright © 2017 Guido Marucci Blas. All rights reserved.
+//
+import UIKit
```
Comment (guidomb) [inline Portal/View/Shadow.swift]: We should not depend on UIKit at this level. UIKit specific should be part of the renderer inside the UIKit directory

File: Portal/View/Shadow.swift [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        self.offset = offset
+        self.radius = radius
+        self.shouldRasterize = shouldRasterize
+    }
+    
+}
+
+public struct Offset {
+    
+    public var posX: Float
```
Comment (guidomb) [inline Portal/View/Shadow.swift]: I would suggest renaming this to `x` and `y`

File: Portal/View/Shadow.swift [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    public var posY: Float
+    
+    internal var asCGSize: CGSize {
```
Comment (guidomb) [inline Portal/View/Shadow.swift]: This should be inside the UIKit directory as an extension or as part of the renderer file

**Review by guidomb — CHANGES_REQUESTED** [review]

File: Portal/View/Component.swift (hunk 1 above)
Comment (cristianames92) [reply Portal/View/Component.swift]: In fact, this is related because when you clip to bounds an image it clips the shadow too, so I add this associated value to configure this parameter. When `clipToBounds` is disabled, the shadow works on an image. If it's always enabled, it will never work directly onto it. I think this has to be on this PR. I will add this to the base stylesheet, because this is a UIView property.

**Review by guidomb — APPROVED** [review]
